anyway/infographics_utils.py

```diff
@@ -1118,22 +1120,53 @@ def localize_items(request_params: RequestParams, items: Dict) -> Dict:
 
 
 @register
+# TODO: As it currently stands, for SubUrbanWidgets such as this, the is_relevant() method is simply is_sub_urban() which checks for
+# the presence of "road1" and "road_segment_name", which do not match the condition of "street1/2_hebrew" having value, which is required
+#  for matching with the "involved_markers_hebrew" table (see SQLAlchemy below). Maybe we should make InjuredAccidentsWithPedestriansWidget
+# inherit Widget instead of SubUrbanWidget?
 class InjuredAccidentsWithPedestriansWidget(SubUrbanWidget):
     name: str = "injured_accidents_with_pedestrians"
 
+    def validate_parameters(self):
+        # TODO: validate each parameter and display message accordingly
+        return self.request_params.location_info.get('yishuv_name') is not None and \
+            (self.request_params.news_flash_obj.street1_hebrew is not None or   \
+            self.request_params.news_flash_obj.street2_hebrew is not None) and  \
+            self.request_params.years_ago is None
+
     def __init__(self, request_params: RequestParams):
         super().__init__(request_params, type(self).name)
+
+        if not self.validate_parameters():
+            logging.exception(f"Could not validate parameters for {NewsFlash} : {request_params.news_flash_obj.id}")
+            return None
+
         self.rank = 18
-        self.text = {"title": "נפגעים הולכי רגל ברחוב ז׳בוטינסקי, פתח תקווה"}
+        self.text = {"title": f"נפגעים הולכי רגל ברחוב(ות) {str.join(', ', [request_params.news_flash_obj.street1_hebrew, request_params.news_flash_obj.street2_hebrew])}, {request_params.news_flash_obj.yishuv_name}"}
 
     @staticmethod
+    # TODO: change?
     def is_in_cache() -> bool:
         return False
 
     def generate_items(self) -> None:
-        self.items = (
-            InjuredAccidentsWithPedestriansWidget.injured_accidents_with_pedestrians_mock_data()
-        )
+        try:
+            query = db.session.query(InvolvedMarkerView)\
+                .with_entities(InvolvedMarkerView.accident_year,    \
+                    InvolvedMarkerView.injury_severity, \
+                    InvolvedMarkerView.injury_severity_hebrew,\
+                    func.count(distinct(f"{InvolvedMarkerView.provider_and_id}_{InvolvedMarkerView.involve_id}")))  \
+                .filter(InvolvedMarkerView.accident_yishuv_name == self.request_params.location_info['yishuv_name'])\
+                .filter(InvolvedMarkerView.injured_type == 1)   \
+                .filter(or_(InvolvedMarkerView.street1_hebrew == self.request_params.news_flash_obj.street1_hebrew,
+                    InvolvedMarkerView.street2_hebrew == self.request_params.news_flash_obj.street2_hebrew))    \
+                .filter(InvolvedMarkerView.accident_year >= datetime.date.today().year - self.request_params.years_ago)   \
```

Comment:
Check out [get_query](https://github.com/hasadna/anyway/blob/0fd4f869f48476bfb85097435c82ab097486584c/anyway/infographics_utils.py#L84) - perhaps we should be consistent in the way we query the timestamps in our data?

Reply:
I think RequestParams already holds the start and end dates that can be used for queries.
In addition check whether it is possible to use `get_accidents_stats()` to do the query.

Reply:
@ziv17 get_accidents_stats does not contain implementation of or_/and_ clauses of SQLAlchemy 
@atalyaalon i can indeed use get_query, but i get a different amount of items as the end_time does not necessarily go all the way to the present - it appears get_latest_accident_date does not return the latest accident date for tables which are not AccidentMarkerView. For example, in the InjuredAccidentsWithPedestriansWidget, where we query against InvolvedMarkerView

Reply:
Can we take today's date as the end date?

Reply:
@shaysw not sure I follow - [see the following query](https://app.redash.io/hasadna/queries/975346/source) for the max timestamps across tables.
Seems there is a problem with the vehicles_markers_hebrew table - I'm checking what's the problem - however this shouldn't concern you at the moment since it's not a table you're using in this Widget.

@ziv17 Regarding the end date - by now we used the end_date as the latest date of the latest accident in the DB.
In the future, there might be a use case in which we'd like to filter dates that do not end in today's date but for example, end in a different date - hence I think that keeping the end_date initialization in the `get_request_params` function and using it across the widget queries is important.

Reply:
@atalyaalon the redash query checks against 3 tables while the get_latest_accident_date is performed only against AccidentMarkerView, therefore there's a mismatch between the latest accident date and what get_latest_accident_date() returns

Reply:
will use same logic as in get_query